File: frontend/src/controllers/SpecialOperationController.ts
import api from '@/controllers/api';
import { AxiosResponse } from 'axios';
import { DragonResultDto } from '@/interfaces/dto/specialoperations/DragonResultDto';
import { AverageAgeDto } from '@/interfaces/dto/specialoperations/AverageAgeDto';


export default class SpecialOperationController {
  static async deepestCaveDragon(): Promise<AxiosResponse<DragonResultDto>> {
    return api.get<DragonResultDto>('/special-operations/deepest-cave-dragon');
  }

  static async averageAge(): Promise<AxiosResponse<AverageAgeDto>> {
    return api.get<AverageAgeDto>('/special-operations/average-age');
  }
}

File: src/main/java/org/itmo/isLab1/specialoperations/SpecialOperationController.java
package org.itmo.isLab1.specialoperations;

import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.itmo.isLab1.specialoperations.dto.*;


@RestController
@RequestMapping(value = "/api/special-operations", produces = MediaType.APPLICATION_JSON_VALUE)
@RequiredArgsConstructor
public class SpecialOperationController {

  private final SpecialOperationService service;

  @GetMapping("/average-age")
  public ResponseEntity<AverageAgeDto> getAverageDragonAge() {
    var averageAge = service.getAverageDragonAge();
    return ResponseEntity.ok(averageAge);
  }

  @GetMapping("/deepest-cave-dragon")
  public ResponseEntity<DragonResultDto> getDragonInDeepestCave() {
    var dragon = service.getDragonInDeepestCave();
    return ResponseEntity.ok(dragon);
  }
}